\section{Conclusions}
This study was conducted as part of a bigger project aiming to develop a pipeline for the analysis of longitudinal RNA-seq data, as generated by HTTr assays. In this regard, one of the main challenges faced by the scientific community is that there is still no consensus on how to conduct this type of analysis, and which tools are best suited in this context. Hence the importance of revising some of the most used and popular tools.

Unlike other tools that perform differential expression (DE) analysis, the focus of this dissertation was DPGP, a tool specifically designed for clustering genomic features over time. This choice was based on three main reasons: 
\begin{enumerate}
    \item It had been decided that the DE step in the pipeline will be performed using \texttt{ImpulseDE2};
    \item DPGP is among the most popular tools, based on an initial review of the Literature;
    \item Finally, DPGP uses Bayesian inference to infer the parameters that fit the data from the data itself by using model agnostic approaches. In this case, DPGP uses a Dirichlet process (DP) to determine the number of clusters, combined with a Gaussian process (GP) for modelling expression levels over time.
\end{enumerate}

This study had two main objectives: 
\begin{enumerate}
    \item Reproduce the original results from the DPGP paper \citep{mcdowellClusteringGeneExpression2018} to assert its ability of generating consistent results, and simultaneously evaluate its assumptions;
    \item Assess DPGP behaviour on simulated datasets with known characteristics and expected outcomes.
\end{enumerate}

Despite minor discrepancies, the results on the biological dataset originally published by \citeauthor{sharmaRosRTranscriptionFactor2012} gave very consistent results (see section \ref{res:geo}), showing that it is sufficient to provide the same parameters and seed to be able to generate the same outcomes.

On the contrary, the analysis conducted on the simulated datasets (see section \ref{res:simulated}), highlighted how the assumptions used to generate the gene expression levels over time may affect the performance of the tool, which can lead to inconclusive interpretations.

Overall, DPGP has proven to be a potentially good addition for an analytical pipeline, and it could be further investigated.

\section{Future directions}
So far DPGP has been used in this study as a stand-alone tool, but it still needs to be evaluated in comparison to other tools with similar scope and characteristics, and in the context of a full pipeline for the analysis of RNA-seq time series data.
For that purpose, many possible combinations of tools will need to be taken into consideration, \textit{e.g.} employing different pre-processing practices, or DE analysis with different tools, etc.

Also, this study did not focus on the output of DPGP and on how it can be used in the downstream steps of a potential pipeline, such as annotations and pathway analysis, in order to extrapolate a biological interpretation from the data.

Finally, DPGP (and the pipeline) has yet to be tested on data produced from HTTr assays like the ones used by Unilever's R\&D department, and how valuable its interpretation is when compared to other ``common'' pipelines that treat time as any other categorical variable. 
